app/Kwality.UVault.QA/Internal/Factories/Application.Token.Manager.Factory.cs

```diff
@@ -0,0 +1,98 @@
+// =====================================================================================================================
+// = LICENSE:       Copyright (c) 2023 Kevin De Coninck
+// =
+// =                Permission is hereby granted, free of charge, to any person
+// =                obtaining a copy of this software and associated documentation
+// =                files (the "Software"), to deal in the Software without
+// =                restriction, including without limitation the rights to use,
+// =                copy, modify, merge, publish, distribute, sublicense, and/or sell
+// =                copies of the Software, and to permit persons to whom the
+// =                Software is furnished to do so, subject to the following
+// =                conditions:
+// =
+// =                The above copyright notice and this permission notice shall be
+// =                included in all copies or substantial portions of the Software.
+// =
+// =                THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// =                EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+// =                OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// =                NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// =                HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// =                WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// =                FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// =                OTHER DEALINGS IN THE SOFTWARE.
+// =====================================================================================================================
+namespace Kwality.UVault.QA.Internal.Factories;
+
+using JetBrains.Annotations;
+
+using Kwality.UVault.Extensions;
+using Kwality.UVault.M2M.Extensions;
+using Kwality.UVault.M2M.Managers;
+using Kwality.UVault.M2M.Models;
+using Kwality.UVault.M2M.Options;
+using Kwality.UVault.M2M.Stores.Abstractions;
+
+using Microsoft.Extensions.DependencyInjection;
+
+internal sealed class ApplicationTokenManagerFactory
+{
+    private readonly IServiceCollection serviceCollection;
+
+    public ApplicationTokenManagerFactory()
+    {
+        this.serviceCollection = new ServiceCollection();
+    }
+
+    public ApplicationTokenManager<TToken, TModel, TKey> Create<TToken, TModel, TKey>()
+        where TToken : TokenModel
+        where TModel : ApplicationModel<TKey>
+        where TKey : IEqualityComparer<TKey>
+    {
+        this.serviceCollection.AddUVault(static (_, options) => options.UseApplicationTokenManagement<TToken, TModel, TKey>(null));
+        this.serviceCollection.AddSingleton<IApplicationTokenStoreStaticTokenGenerator<Model, TModel, TKey>, ApplicationStoreStaticTokenGenerator<TModel, TKey>>();
```

Comment:
This is an approach which I don't like.
So if you're using the `static` store, you need to register another token generator.
I would keep it plain simple. Let the static store return a simple access token (but not a Guid since a consumer would need to be able to validate it).

I propose that you return a string in the following format "token|expires_in|token_type".
The token itself can be a string, a guid or anything.
This way, when anyone is using the static store, you know what kind of tokens you can expect, and validating these tokens is pretty straight forward.

In case anyone doesn't like this implementation, a custom implementation of the store can be developed.